lib/core/utils/valid-langs.js

```diff
@@ -1,4 +1,71 @@
+
+/**
+ * Determine if a string is a valid language code
+ * @method isValidLang
+ * @memberof axe.utils
+ * @param {String} lang String to test if a valid language code
+ * @returns {Boolean}
+ */
+function isValidLang(lang) {
+	let array = langs;
+
+	// padEnd is not supported in IE11
+	while (lang.length < 3) {
+		lang += '`';
+	}
+	const chars = lang.split('');
+
+	for (let i = 0; i <= chars.length - 1; i++) {
+		const index = chars[i].charCodeAt(0) - 96;
+		array = array[index];
+		if (!array) {
+			return false;
+		}
+	}
+
+	return true;
+}
+
+/**
+ * Returns array of valid language codes
+ * @deprecated
+ * @method validLangs
+ * @memberof axe.utils
+ * @return {Array<Sting>} Valid language codes
+ */
+export function validLangs(langArray) {
+	// account for our external API tests passing non-array things
+	langArray = Array.isArray(langArray) ? langArray : langs;
+
+	let codes = [];
+	langArray.forEach((lang, index) => {
+		const char = String.fromCharCode(index + 96).replace('`', '');
+		if (Array.isArray(lang)) {
+			codes = codes.concat(validLangs(lang).map(newLang => char + newLang));
+		} else {
+			codes.push(char);
+		}
+	});
+
+	return codes;
+}
+
+export default isValidLang;
+
+/*
+If we need to edit the list of langs, use the code below and edit the array of langs:
```

Comment:
I think we should get rid of the actual list and just point to the registry:
https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry

You may want to rerun this on the latest version of the registry, so that we have an up to date version. We haven't updated this list in several years.